utils/firestore.js

```diff
@@ -1406,15 +1425,77 @@ const shipKits = async (data) => {
                 kit_status: "shipped",
                 pickup_date: data.pickup_date,
                 confirm_pickup: data.confirm_pickup
-
             })
+            await kitStatusCounterVariation('shipped', 'assigned');
         return true;
         }
     catch(error){
         return new Error(error);
     }
 }
 
+const storePackageReceipt =  (data) => {
+    if (data.scannedBarcode.length <= 12) {  
+        const response = setPackageReceiptFedex(data);
+        return response;
+    } else { 
+        const response = setPackageReceiptUSPS(data);
+        return response; 
+    }
+
+} 
+
+const setPackageReceiptUSPS = async (data) => {
+    try {
+        const snapshot = await db.collection("participantSelection").where('usps_trackingNum', '==', parseInt(data.scannedBarcode)).get();
+        const docId = snapshot.docs[0].id;
+        await db.collection("participantSelection").doc(docId).update(
+        { 
+            baseline: data
+        })
+            return true;
+        }
+    catch(error){
+        return new Error(error);
+    }
+}
+
+const setPackageReceiptFedex = async (data) => {
+    try {
+        const snapshot = await db.collection("boxes").where('959708259', '==', data.scannedBarcode).get();
+        const docId = snapshot.docs[0].id;
+        await db.collection("boxes").doc(docId).update({ 
+             baseline: data
+        })
+            return true;
+         }
+    catch(error){
+        return new Error(error);
+    }
+}
+
+const kitStatusCounterVariation = async (currentkitStatus, prevKitStatus) => {
+    try {
+        await db.collection("bptlMetrics").doc('--metrics--').update({ 
+            [currentkitStatus]: increment
+        })
+        await db.collection("bptlMetrics").doc('--metrics--').update({ 
+            [prevKitStatus]: decrement
```

Comment:
where are increment and decrement defined?

Reply:
https://github.com/episphere/connectFaas/pull/94/files/71a08e3a8dcbca30f20c8c072c3fe56fc80b2167#diff-be91e66430aa4dd2f61668d1bd17d3cb449f57947e13bfa2f6bc087919967282R11